src/main/java/io/github/terra121/util/AzimuthUtils.java

```diff
@@ -0,0 +1,35 @@
+package io.github.terra121.util;
+
+public class AzimuthUtils {
+	public static String azimuthToFacing(float azimuth) {
+		if (azimuth > 270 && azimuth < 360) {
+			return "Northwest";
+		}
+		else if (azimuth > 180 && azimuth < 270) {
+			return "Southwest";
+		}
+		else if (azimuth > 90 && azimuth < 180) {
+			return "Southeast";
+		}
+		else if (azimuth > 0 && azimuth < 90) {
+			return "Northeast";
+		}
+		else if (azimuth == 0) {
+			return "North";
+		}
+		else if (azimuth == 270) {
+			return "West";
+		}
+		else if (azimuth == 180) {
+			return "South";
+		}
+		else if (azimuth == 90) {
+			return "East";
```

Comment:
none of these cases (north, west, south, east) are likely to ever actually happen, because the floating-point result is extremely unlikely to ever actually be an exact value. i would recommend you have each direction string correspond to a 45° segment (so east would be 67.5° to 112.5°, etc.)

Reply:
Agreed, those are equal cases are irrelevant.